internal/printer/print-to-tsx.go

```diff
@@ -67,8 +67,9 @@ func isValidTSXAttribute(a Attribute) bool {
 		if i != 0 && !(isValidFirstRune(ch) ||
 			unicode.In(ch, unicode.Mn, unicode.Mc, unicode.Nd, unicode.Pc)) &&
 			// : is allowed inside TSX attributes, for namespaces purpose
+			// - is allowed inside TSX attributes, for custom attributes
 			// See https://facebook.github.io/jsx/#prod-JSXNamespacedName
-			ch != ':' {
+			ch != ':' && ch != '-' {
```

Comment:
I misread the JSX spec when implementing this, I didn't realise that it was saying that `-` was a valid character, I thought it was part of the visual design of the page 🤦‍♀️